portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,18 +40,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = getCommenterName(request);
-    comments.add(name);
-    // String comment = 
+    String name = getRequestParam(request, "custom");
+    String comment = getRequestParam(request, "text-input");
+    comments.add(name + " said: " + comment);
+    
     response.sendRedirect("/comments.html");
   }
 
-  private String getCommenterName(HttpServletRequest request) {
-    String name = request.getParameter("custom");
-    if (name == null) {
-      name = "Anonymous";
+  private String getRequestParam(HttpServletRequest request, String inputName) {
+    String input = request.getParameter(inputName);
+
+    if (input == null) {
```

Comment:
Consider returning early to reduce nesting. Deep nesting is generally less clear to read:
https://dzone.com/articles/code-smells-deeply-nested-code

if (input != null) {
   return input;
}

if (inputName.equals("custom")) {
        input = "Anonymous";
      } else {
        input = "";
      }
return input